pkg/artifactory/resource_artifactory_virtual_repository.go

```diff
@@ -157,6 +157,11 @@ func resourceVirtualRepositoryRead(d *schema.ResourceData, m interface{}) error
 	c := m.(*ArtClient).ArtOld
 
 	repo, resp, err := c.V1.Repositories.GetVirtual(context.Background(), d.Id())
+
+	if resp == nil {
```

Comment:
Check `err` first. If you check error first you probably never need to check if `resp` is `nil`.